concordium-sdk/src/main/java/com/concordium/sdk/Client.java

```diff
@@ -313,9 +311,28 @@ public Semver getVersion() {
         return new Semver(versionString);
     }
 
+    /**
+     * Get the list of identity providers in the given block.
+     *
+     * @param blockHash Block {@link Hash}.
+     * @return {@link ImmutableList} of {@link IpInfo}.
+     */
+    public ImmutableList<IpInfo> getIdentityProviders(Hash blockHash) throws Exception {
+        val res = server()
+                .getIdentityProviders(ConcordiumP2PRpc.BlockHash.newBuilder().setBlockHash(blockHash.asHex()).build());
+
+        val array = IpInfo.fromJsonArray(res);
+
+        if (Objects.isNull(array)) {
+            throw new Exception(String.format("Could not get Identity Providers at block %s", blockHash.asHex()));
```

Comment:
I think it's safe to say that if the exception is triggered then it's because the block hash did not match a block on chain, thus I think you should throw a `BlockNotFoundException` here.

Reply:
done